packages/flutter_tools/lib/src/commands/precache.dart

```diff
@@ -66,6 +66,35 @@ class PrecacheCommand extends FlutterCommand {
     ]
   };
 
+  /// Returns a reverse mapping of _expandedArtifacts, from child artifact name
+  /// to umbrella name.
+  Map<String, String> get _umbrellaForArtifactMap {
```

Comment:
Since both of these do non-trivial calculations, I would make them methods like `_getUmbrellaForArtifactMap()`

Reply:
You could also use a `for` element instead, if you'd like

```
final Map<String, String> reverseMap = <String, String>{
  for (MapEntry<String, List<String>> entry in _expandedArtifacts.entries)
    for (String child in entry.value)
      childArtifactName: entry.key
};
```